### DISCUSSION TITLE
How to read the difference in 2 images, using COM/DLL version?

### DISCUSSION THREAD

**Discussion by FrontierDK** [ImageMagick/ImageMagick#3261] [Help]
Using the DLL version of ImageMagick, I'm trying to the find the difference in percentage from 2 images. The following VBscript runs, but sends the difference to the console, rather than returning the difference to the variable.

Any chance of a fix?

```
option explicit
dim IM, result
set IM=wscript.CreateObject("ImageMagickObject.MagickImage.1")
result=Cint(IM.compare("-metric", "AE", "picture1.png", "picture2.png", "NULL:"))
set IM=nothing
wscript.echo vbcrlf + Cstr(result)
```

**Comment by snibgo** [comment]
"compare" writes the text to stderr, not stdout. I don't know how to modify your code to capture stderr output. It can be done like this:
```
Dim wsh
Set wsh = CreateObject("Wscript.Shell")

dim command
command = "magick compare -metric AE toes.png t.png NULL:"

dim objExec, strf
Set objExec = wsh.Exec(command)
strf = objExec.StdErr.Readline
wscript.echo vbcrlf + "strf is " + Cstr(strf)
```

**Comment by FrontierDK** [comment]
Hi Snibo,

I was hoping to avoid the shell, but your script does work - thank you for sharing :)

**Comment by snibgo** [comment]
How about:
```
result2=IM.convert("toes.png", "t.png", "-metric", "RMSE", "-format", "%[distortion]", "-compare", "info:")
wscript.echo vbcrlf + "result2 is " + Cstr(result2)
```

**Reply by FrontierDK** [reply]
You, sir, are a genious :) :) :) 👍
